### ISSUE TITLE
Can we enable shallow cloning?

### ISSUE THREAD

**Issue by BenSandeen** [CircleCI-Public/path-filtering-orb#71]
## Describe Request:
The code checkout step of this orb is the vast majority of the build time for this step in our CircleCI workflow.  Shallow checkout (ie: using `git clone --depth=1` would cut this step's duration by roughly 75% for our monolith codebase
## Examples:
I don't know enough about how this orb works to provide code examples, but I _may_ be willing to learn if it's not too involved
## Supporting Documentation Links:
Here's our current task that uses this orb:
<img width="1171" alt="Screenshot 2023-04-14 at 10 27 02 AM" src="https://user-images.githubusercontent.com/12025856/232087560-1ba3a0e5-b035-4c42-932d-7f3fa5ce6d14.png">

We use this `shallow_checkout` orb (https://circleci.com/developer/orbs/orb/datacamp/shallow-checkout) for the following steps of this build, and it routinely runs the code checkout step in 9-10 seconds, which would be a great benefit to speed up our builds a bit

**Comment by eli-adept** [comment]
Also interested in this. There seems to be an old closed PR that implemented this using another orb

https://github.com/CircleCI-Public/path-filtering-orb/pull/48/files

I'm happy to revive that attempt or hack together another solution if the maintainers have any appetite for this change

**Comment by Fernando-Abreu** [comment]
I will add this to our backlog.

**Comment by BenSandeen** [comment]
Is there any chance we could just merge that PR?  Do we know why @kynefuk closed it?

**Comment by krzema12** [comment]
Our repo takes over a minute to clone entirely. The tip of it is pretty lightweight now, clones within several seconds. Given that our CI worfklow uses path filtering, and then in other jobs we need to clone the repo again, it currently takes 2 extra minutes that we could save on cloning. While we have control over what happens in the other jobs, we can just optimize them. The initial step using the path-filtering orb is not possible to optimize (at least without rewriting it) because this ticket is still not addressed.

Please consider implementing this simple feature that will save compute resources (also your workers!) and improve productivity of multiple engineers.

**Comment by sebastian-lerner** [comment]
Hey folks, we are testing out a "blobless clone" which we hope will achieve a similar outcome as a shallow clone.  If you would like to be included in early testing, email me at sebastian@circleci.com

**Comment by edoardo-bluframe** [comment]
Hey everybody

We **really** need this over here - like 2 min 45 sec each time of extra build when all we need is shallow??

This is a **really** cool orb but we **do** need that feature

Is there a way we can help?

Maybe we can have a look at that PR from @kynefuk?

Create a new one?

What kinda support do you need from the community to make this happen?? :smile:

**Comment by mscrivo** [comment]
We forked this orb and made one that supports shallow cloning if folks are interested: https://circleci.com/developer/orbs/orb/affinity/path-filtering-orb. It's very similar to https://github.com/CircleCI-Public/path-filtering-orb/pull/48, but that PR was broken and didn't actually work. We've been using this for the past month or so and it's been working very well.

**Comment by edoardo-bluframe** [comment]
> We forked this orb and made one that supports shallow cloning if folks are interested: https://circleci.com/developer/orbs/orb/affinity/path-filtering-orb. It's very similar to [#48](https://github.com/CircleCI-Public/path-filtering-orb/pull/48), but that PR was broken and didn't actually work. We've been using this for the past month or so and it's been working very well.

We wanna try it!

Does it work **exactly** the same? We just change the orb import?

**Comment by mscrivo** [comment]
> Does it work **exactly** the same? We just change the orb import?

yep, just replace the circleci/path-filtering orb with: `path-filtering: affinity/path-filtering-orb@0.9.1` in your orbs list and then add the shallow clone options to your path-filtering workflow, for example:

```
- path-filtering/filter:
    base-revision: master
    config-path: .circleci/continue_config.yml
    enable_shallow_checkout: true
    checkout_shallow_depth: 200 # adjust as necessary 
    checkout_shallow_fetch_depth: 200 # adjust as necessary
```

**Comment by Peter-Darton-i2** [comment]
Doesn't the [blobless clone](https://github.com/CircleCI-Public/path-filtering-orb/issues/71#issuecomment-1910557030) feature (mentioned above) eliminate the need for this entirely?

**Comment by Peter-Darton-i2** [comment]
> It might? But it's not available yet.

Oh dear.
We had been informed that it was all sorted last year as it's been something we've been asking for ever since we started using CircleCI; we were never informed of any rollback.
... in fact the [official documentation](https://circleci.com/docs/configuration-reference/#checkout) still says blobless clones are happening; no mention of a rollback ☹ 

That said, it's not a total shock that there were "issues"; it is a "very clever" solution and (in my experience) anything "clever" that is very different to "_what everyone else does_" is rarely an unqualified success (otherwise "everyone else" would be doing it too), but it is sad that it was announced with great fanfare and then silently withdrawn (almost as sad, in fact, as the lack of shallow cloning support in the `checkout` command itself)

I guess this will have to go back on our "monthly discussions" agenda.

**Comment by edoardo-bluframe** [comment]
> > Does it work **exactly** the same? We just change the orb import?
> 
> yep, just replace the circleci/path-filtering orb with: `path-filtering: affinity/path-filtering-orb@0.9.1` in your orbs list and then add the shallow clone options to your path-filtering workflow, for example:
> 
> ```
> - path-filtering/filter:
>     base-revision: master
>     config-path: .circleci/continue_config.yml
>     enable_shallow_checkout: true
>     checkout_shallow_depth: 200 # adjust as necessary 
>     checkout_shallow_fetch_depth: 200 # adjust as necessary
> ```

Life savers!! We just tried it and went from 2-3 minutes average to **20 seconds** checkout!

And we haven't adjusted default values - outta the box it saves us 2 min per build

You guys are **terrific!!** 🎉

**Comment by anthony-j-castro** [comment]
Hi everyone, a couple of updates here. We just published [version 2.1.0](https://github.com/CircleCI-Public/path-filtering-orb/releases/tag/v2.1.0) which exposes the checkout method as a parameter, allowing you to specify a blobless clone (more information in our [docs](https://circleci.com/docs/reference/configuration-reference/#checkout)). Right now, the default is a full clone to match existing behavior.

We'll publish a new major release of this orb on November 3 to make blobless the default method to align with the rollout that was [recently announced](https://circleci.com/changelog/default-method-used-to-checkout-code-from-your-repository-is-changing-on-nov/).

And back to the original request: shallow clones are still in the works. We'll provide an update here when that functionality is available in this orb.

**Comment by BenSandeen** [comment]
@anthony-j-castro One of my coworkers brought this to my attention last week!  It's exciting to see; it appears to help our build times nearly as much as a shallow clone; thanks!  🤘
